### PR TITLE
add tags to support backward compatibility

### PR DESCRIPTION
**Pull request by eabdelmaqsod**
Hello @Geartrixy ,

This pull request is to add tags for IGW, NatGW and default route tables to support backward compatibility.

Please review, merge and create the correct tag version

Best Regards,
Enas Abdelmaqsod

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -41,4 +41,14 @@ locals {
   sn_public_a = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_a) > 0 ? length(var.public_subnets_a) : 0))
   sn_public_b = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_b) > 0 ? length(var.public_subnets_b) : 0))
   sn_public_c = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_c) > 0 ? length(var.public_subnets_c) : 0))
+
+  #igw_tags
+  igw_tags =  merge( {  "Name" = var.igw_tags["Name"] }, var.igw_tags)
```
Comment (Geartrixy) [inline locals.tf]: @eabdelmaqsod 
igw_tags = merge({"Name" = var.igw_name}, var.igw_tags)

+ declare "var.igw_name" (default = aws_vpc.main.id)

File: locals.tf [hunk 2]
```diff
@@ -41,4 +41,14 @@ locals {
   sn_public_a = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_a) > 0 ? length(var.public_subnets_a) : 0))
   sn_public_b = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_b) > 0 ? length(var.public_subnets_b) : 0))
   sn_public_c = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_c) > 0 ? length(var.public_subnets_c) : 0))
+
+  #igw_tags
+  igw_tags =  merge( {  "Name" = var.igw_tags["Name"] }, var.igw_tags) 
+
+  #nat_gw_tags
+  nat_gw_tags = merge({  "Name" = var.nat_gw_tags["Name"] }, var.nat_gw_tags)
```
Comment (Geartrixy) [inline locals.tf]: @eabdelmaqsod 
nat_gw_tags = merge({"Name" = var.nat_gw_name}, var.nat_gw_tags)

+ declare "var.nat_gw_name" (default = aws_vpc.main.id)

**Review by Geartrixy — COMMENTED** [review]
@eabdelmaqsod please amend as specified in comments

File: locals.tf (hunk 1 above)
Comment (eabdelmaqsod) [reply locals.tf]: @Geartrixy ,

There is no need to specify a new var for name to just set default value, I set the local value to vpc_name if the no tags are passed

File: locals.tf (hunk 2 above)
Comment (eabdelmaqsod) [reply locals.tf]: @Geartrixy ,

There is no need to specify a new var for name to just set default value, I set the local value to vpc_name if the no tags are passed

**Pull request merged**